### PR TITLE
[finishes #163217645] implented feedback ,used blueprints,modified te…

### PR DESCRIPTION
**Pull request by nyambaka**
**IMPLENTED FEEDBACK**

1.used the flask internal error handling methods
2.used blueprints to version my application
3.reverted back to original flask routing system
4.ensured that the correct status code is returned in error messages

**How to test**
for now you can clone my repo https://github.com/nyambaka later after merging
This application has been hosted on heroku https://firstquestioner.herokuapp.com/
you can issue normal request adding /api/v1 before each 

this is an implementation of pivotal tracker stories hosted at https://www.pivotaltracker.com/n/projects/2235577
using the id 163217645 you can read the story in details

### CODE REVIEW

File: app/api/test_api/test_a_user_registration.py [hunk 1]
```diff
@@ -1,122 +1,107 @@
...
         self.app = application.test_client()
 
     def test_a_signup_with_wrong_data(self):
         global data
-        response = self.app.post(self.url_prefix+"/signup",headers ={"Content-Type":"application/json"},data = "")
-        self.assertEqual(response.status_code,200,"signup method is allowed")
-        self.assertEqual(json.loads(response.get_data().decode()),{'error': 'this route required proper formated json '},"correct error message should be given back")
+        response = self.app.post(self.url_prefix + "/signup", headers={"Content-Type": "application/json"}, data="")
+        self.assertEqual(response.status_code, 400, "signup method is allowed")
+        self.assertEqual(json.loads(response.get_data().decode()),{'error': 'this route required proper formated json ', 'status': 400}
```
Comment (dessHub) [inline app/api/test_api/test_a_user_registration.py:21]: Break this line. Have `json.loads()` on its line and assign a variable.

File: app/api/test_api/test_a_user_registration.py [hunk 2]
```diff
@@ -1,122 +1,107 @@
...
+        self.url_prefix = "api/v1"
         self.app = application.test_client()
 
     def test_a_signup_with_wrong_data(self):
         global data
-        response = self.app.post(self.url_prefix+"/signup",headers ={"Content-Type":"application/json"},data = "")
-        self.assertEqual(response.status_code,200,"signup method is allowed")
-        self.assertEqual(json.loads(response.get_data().decode()),{'error': 'this route required proper formated json '},"correct error message should be given back")
+        response = self.app.post(self.url_prefix + "/signup", headers={"Content-Type": "application/json"}, data="")
+        self.assertEqual(response.status_code, 400, "signup method is allowed")
```
Comment (dessHub) [inline app/api/test_api/test_a_user_registration.py:20]: You are passing status code 400 but the assertion message is saying `Signup method is allowed`

File: app/api/test_api/test_a_user_registration.py (hunk 1 above)
Comment (nyambaka) [reply app/api/test_api/test_a_user_registration.py:21]: thanks will certainly check it

File: app/api/test_api/test_a_user_registration.py [hunk 3]
```diff
@@ -1,122 +1,107 @@
...
+        self.data = {"password": 1}
 
     def test_c_test_signup_with_no_first_name(self):
-        response = self.app.post(self.url_prefix + "/signup", headers={"Content-Type": "application/json"},
-                                 data=json.dumps(data))
-        self.assertEqual(json.loads(response.get_data().decode())["error"], 'first name has not been set',
-                         "password missing error shoulb be noted")
+        response= self.tst_zone()
+        self.assert_master(response, 403, "password missing error shoulb be noted")
         data["firstname"] = 4
```
Comment (dessHub) [inline app/api/test_api/test_a_user_registration.py:42]: Is this relevant?

File: app/api/test_api/test_a_user_registration.py [hunk 4]
```diff
@@ -1,122 +1,107 @@
...
 
     def test_d_with_in_correct_first_name(self):
-        response = self.app.post(self.url_prefix + "/signup", headers={"Content-Type": "application/json"},
-                                 data=json.dumps(data))
-        self.assertEqual(json.loads(response.get_data().decode())["error"], 'first name should be of type string',
-                         "password missing error shoulb be noted")
+        response= self.tst_zone()
+        self.assert_master(response, 404, "password missing error shoulb be noted")
         data["firstname"] = "nelson"
```
Comment (dessHub) [inline app/api/test_api/test_a_user_registration.py:47]: Is this relevant?

File: app/api/test_api/test_a_user_registration.py [hunk 5]
```diff
@@ -1,122 +1,107 @@
...
 
     def test_e_with_last_name(self):
-        response = self.app.post(self.url_prefix + "/signup", headers={"Content-Type": "application/json"},
-                                 data=json.dumps(data))
-        self.assertEqual(json.loads(response.get_data().decode())["error"], 'last name has not been set',
-                         "password missing error shoulb be noted")
+        response= self.tst_zone()
+        self.assert_master(response, 405, "password missing error shoulb be noted")
         data["lastname"] = 1
```
Comment (dessHub) [inline app/api/test_api/test_a_user_registration.py:52]: Is this relevant?

File: app/api/test_api/test_a_user_registration.py [hunk 6]
```diff
@@ -1,122 +1,107 @@
...
 
     def test_f_with_incorrect_last_name(self):
-        response = self.app.post(self.url_prefix + "/signup", headers={"Content-Type": "application/json"},
-                                 data=json.dumps(data))
-        self.assertEqual(json.loads(response.get_data().decode())["error"], 'last name should be of type string',
-                         "password missing error shoulb be noted")
+        response= self.tst_zone()
+        self.assert_master(response, 406, "password missing error shoulb be noted")
         data["lastname"] = "nyambaka"
```
Comment (dessHub) [inline app/api/test_api/test_a_user_registration.py:57]: Why is this here?

File: app/api/test_api/test_api_v1.py [hunk 7]
```diff
@@ -1,39 +1,47 @@
...
+    def tst_zone(self):
+        response = self.client.post(self.url_prefix + "/meetups", headers={"Content-Type": "application/json"},
+                                 data=json.dumps(data))
+        return response
+
+    def asset_master(self,response,error_code):
+        self.assertEqual(json.loads(response.get_data().decode())["error"], en_errors[error_code],
+                         "passing in empty json object")
+
+    def test_b_without_l(self):
```
Comment (dessHub) [inline app/api/test_api/test_api_v1.py:33]: What exactly are you testing here? Use descriptive naming conventions.

**Comment by dessHub** [comment]
Good work implementing feedback. However there are a number of things you need to take care of. 
1. Are the lines appearing after assertions relevant?
2. Why are your assertion messages one word? Should be a string of words.
3. Use descriptive namings.

**Comment by nyambaka** [comment]
Those lines are very relevant to my tests ..because for example I check the error that will occur if one does not provide the password then later after asserting I add that password ...so that next test will check if password is in correct format..my asserting messages are stored in a particular particular dictionary those ints represent the keys to those error messages ..

**Comment by nyambaka** [comment]
Thanks for taking time to review my code
